File: tape.c

#include <stdio.h>
#include <stdlib.h>

int main()
{
    //declare variables
    int motor, forward, reverse, stop, read, write, erase, position;
    
    //initialize variables
    motor = 0;
    forward = 0;
    reverse = 0;
    stop = 0;
    read = 0;
    write = 0;
    erase = 0;
    position = 0;
    
    //read in user input
    printf("Enter 1 to control the motor, 2 to move forward, 3 to move in reverse, 4 to stop, 5 to read from the tape, 6 to write to the tape, 7 to erase a section of the tape from position counter A to B, 8 to read the current tape counter position: ");
    scanf("%d", &motor);
    
    //if statements to control tape drive
    if (motor == 1)
    {
        printf("The motor is now on.\n");
    }
    
    if (forward == 2)
    {
        printf("The tape is now moving forward.\n");
    }
    
    if (reverse == 3)
    {
        printf("The tape is now moving in reverse.\n");
    }
    
    if (stop == 4)
    {
        printf("The tape has stopped.\n");
    }
    
    if (read == 5)
    {
        printf("The tape is now being read.\n");
    }
    
    if (write == 6)
    {
        printf("The tape is now being written to.\n");
    }
    
    if (erase == 7)
    {
        printf("A section of the tape is now being erased from position counter A to B.\n");
    }
    
    if (position == 8)
    {
        printf("The current tape counter position is being read.\n");
    }
    
    return 0;
}

File: tape.pas
 
program TapeDrive;

var
 Motor, Forward, Reverse, Stop, Read, Write, Erase, Position : integer;

begin
 Motor := 1;
 Forward := 2;
 Reverse := 3;
 Stop := 4;
 Read := 5;
 Write := 6;
 Erase := 7;
 Position := 8;

 {Control the motor}
 if Motor = 1 then
  begin
   {Turn the motor on}
   writeln('The motor is on');
  end
 else if Motor = 0 then
  begin
   {Turn the motor off}
   writeln('The motor is off');
  end;

 {Control the direction}
 if Forward = 1 then
  begin
   {Move the tape forward}
   writeln('The tape is moving forward');
  end
 else if Reverse = 1 then
  begin
   {Move the tape in reverse}
   writeln('The tape is moving in reverse');
  end;

 {Stop the tape}
 if Stop = 1 then
  begin
   {Stop the tape}
   writeln('The tape has stopped');
  end;

 {Read from the tape}
 if Read = 1 then
  begin
   {Read from the tape}
   writeln('Reading from the tape');
  end;

 {Write to the tape}
 if Write = 1 then
  begin
   {Write to the tape}
   writeln('Writing to the tape');
  end;

 {Erase a section of the tape}
 if Erase = 1 then
  begin
   {Erase from position A to B}
   writeln('Erasing from position A to B');
  end;

 {Read the current tape position}
 if Position = 1 then
  begin
   {Read the current tape position}
   writeln('The current tape position is');
  end;
end.
